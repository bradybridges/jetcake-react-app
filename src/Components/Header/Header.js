import React from 'react';
import './Header.scss';
import { Link } from 'react-router-dom';

export const Header = ({toggleShowLogin, user, handleLogout}) => {
  return (
    <header>
      <h1 id="main-header">Hero Page</h1>
      <nav>
        <ul id="nav-ul">
          <li className="nav-link"><span className="link" onClick={toggleShowLogin}>Login</span></li>
          <li className="nav-link"><Link className="link" to="/create">Create Account</Link></li>
          <li className="nav-link"><Link className="link" to="/profile">View Profile</Link></li>
          { user && <span className="link" onClick={handleLogout}>Logout</span>}
        </ul>
      </nav>
    </header>
  );
}