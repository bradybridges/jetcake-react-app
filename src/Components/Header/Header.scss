@import "../base";

header {
  height: 15vh;
  display: flex;
  justify-content: space-between;
  align-items: center;
  box-shadow: $primaryBoxShadow;
  background-color: $primaryBgColor;
  &:hover {
    cursor: default;
  }
}

#main-header {
  @include primaryHeader();
}

#account-container {
  width: 30%;
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  justify-content: space-evenly;
  height: 100px;
  padding-right: 5%;
}

.account-btn {
  @include button;
  display: flex;
  justify-content: center;
  align-items: center;
}

@media only screen and (max-width: 900px) {
  #account-container {
    width: 50%;
  }

  #main-header {
    width: 50%;
  }
}

@media only screen and (max-width: 450px) {
  #main-header {
    font-size: 2.5em;
  }
}