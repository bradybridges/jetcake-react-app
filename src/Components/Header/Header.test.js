import React from 'react';
import { shallow } from 'enzyme';
import { Header } from './Header';

describe('Header', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<Header
      toggleShowLogin={jest.fn()}
      user={{ email: 'test@test.com' }}
      handleLogout={jest.fn()}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});

// toggleShowLogin={this.toggleShowLogin} user={this.state.user} handleLogout={this.handleLogout}
