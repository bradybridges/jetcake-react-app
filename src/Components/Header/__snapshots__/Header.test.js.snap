// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
<header>
  <Link
    id="main-header"
    to="/"
  >
    Hero Page
  </Link>
  <section
    id="account-container"
  >
    <Link
      className="account-btn"
      to="/profile"
    >
      View Profile
    </Link>
    <button
      className="account-btn"
      onClick={[MockFunction]}
    >
      Logout
    </button>
  </section>
</header>
`;

exports[`Header should not show navigation when showNav is false 1`] = `
<header>
  <Link
    id="main-header"
    to="/"
  >
    Hero Page
  </Link>
  <section
    id="account-container"
  />
</header>
`;

exports[`Header should show appropriate navigation when user is not signed in 1`] = `
<header>
  <Link
    id="main-header"
    to="/"
  >
    Hero Page
  </Link>
  <section
    id="account-container"
  >
    <button
      className="account-btn"
      onClick={[Function]}
    >
      Login
    </button>
    <button
      className="account-btn"
      onClick={[Function]}
    >
      Create Account
    </button>
  </section>
</header>
`;
